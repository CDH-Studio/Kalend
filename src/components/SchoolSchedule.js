import React from 'react';
import {ImageBackground, StatusBar, StyleSheet, View, Image, Text, Platform, TouchableOpacity} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';

class SchoolSchedule extends React.Component {

    static navigationOptions = {
        title: 'Add School Schedule',
        headerTintColor: '#fff',
        headerTitleStyle: {
          fontWeight: 'bold',
          fontFamily: 'Raleway-Regular'
        },
        headerTransparent: true,
        headerStyle: {
          backgroundColor: 'rgba(11, 63, 126, 0.4)',
          marginTop: Platform.OS === 'ios' ? 0 : StatusBar.currentHeight
        }
      };
    


    render() {
        return (
            <LinearGradient style={styles.container} colors={['#1473E6', '#0E55AA']}>
                <ImageBackground style={styles.container} source={require('../assets/img/loginScreen/backPattern.png')} resizeMode="repeat">
                    <StatusBar translucent={true} backgroundColor={'#00000050'} />
                    
                    <View style={styles.content}>
                        <View style={styles.instruction}>
							<Image style={styles.schoolIcon} source={require('../assets/img/schoolSchedule/school.png')} resizeMode="contain" />
							<Text style={styles.text}>Import your school schedule by importing or taking a picture</Text>
						</View>

                        <View>
                            <TouchableOpacity style={styles.buttonSelect} onPress={() => this.props.navigation.navigate('SchoolScheduleSelectPicture')}>
                                <Text style={styles.buttonSelectText}>SELECT A PICTURE</Text>
                            </TouchableOpacity>

                            <TouchableOpacity style={styles.buttonTake} onPress={() => this.props.navigation.navigate('SchoolScheduleTakePicture')}>
                                <Text style={styles.buttonTakeText}>TAKE A PICTURE</Text>
                            </TouchableOpacity>
                        </View>

                        <View style={styles.section}>
                            <View style={styles.sectionIconRow}>
                                <Image style={styles.sectionIcon} source={require('../assets/img/schoolSchedule/sectionActive.png')} resizeMode="contain" />
                                <Image style={styles.sectionIcon} source={require('../assets/img/schoolSchedule/sectionInactive.png')} resizeMode="contain" />
                                <Image style={styles.sectionIcon} source={require('../assets/img/schoolSchedule/sectionInactive.png')} resizeMode="contain" />
                                <Image style={styles.sectionIcon} source={require('../assets/img/schoolSchedule/sectionInactive.png')} resizeMode="contain" />
                                <Image style={styles.sectionIcon} source={require('../assets/img/schoolSchedule/sectionInactive.png')} resizeMode="contain" />
                            </View>
                            <View>
                                <TouchableOpacity style={styles.skipButton}>
                                    <Text style={styles.skipButtonText}>Skip</Text>
                                </TouchableOpacity>
                            </View> 
                        </View>
                    </View>
                </ImageBackground>
            </LinearGradient>
        )
    }
}

const styles = StyleSheet.create({
	container: {
		flex: 1,
		width: '100%',
		height: '130%' //Fixes pattern bug
	},

	content: {
		alignItems: 'center',
		flex: 1,
		flexDirection: 'column',
        justifyContent: 'space-between',
        marginTop: 160
    },

    schoolIcon: {
		height: 130,
		width: 130
	},
    
    instruction: {
        flexDirection: 'row',
        justifyContent: 'center'
    },

	text: {
		fontFamily: 'Raleway-Regular',
		color: '#FFFFFF',
		fontSize: 20,
        paddingTop: 30,
        paddingLeft: 15,
		textShadowColor: 'rgba(0, 0, 0, 0.40)',
		textShadowOffset: { width: -1, height: 1 },
        textShadowRadius: 20,
        width: 220
    },
    
    buttonSelect: {
        borderRadius: 12,
        backgroundColor: '#FFFFFF',
        padding: 17,
        alignItems: 'center',
        width: 300
    },

    buttonSelectText: {
        fontFamily: 'Raleway-Regular',
        fontSize: 15,
        color: '#1473E6'
    },

    buttonTake: {
        borderRadius: 12,
        backgroundColor: 'transparent',
        borderWidth: 3,
        borderColor: '#FFFFFF',
        padding: 17,
        alignItems: 'center',
        marginTop: 20,
        width: 300
    },

    buttonTakeText: {
        fontFamily: 'Raleway-Regular',
        fontSize: 15,
        color: '#FFFFFF'
    },

    section: {
        flexDirection: 'row',
        justifyContent: 'center',
        marginTop: 50,
        marginBottom: 10
    },

    sectionIconRow: {
        flexDirection: 'row'
    },

    sectionIcon: {
        width: 20,
        height: 20,
        margin: 8
    },

    skipButton: {
        marginTop: 5,
        marginLeft: 20,
        justifyContent: 'flex-end'
    },

    skipButtonText: {
        color: 'white',
        fontFamily: 'Raleway-Regular',
        fontSize: 15
    }
});

export default SchoolSchedule;