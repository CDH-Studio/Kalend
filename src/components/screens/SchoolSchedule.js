import React from 'react';
import { StatusBar, View, Text, Platform, TouchableOpacity } from 'react-native';
import FontAwesome5 from 'react-native-vector-icons/FontAwesome5';
import { SchoolScheduleSelectPictureRoute, SchoolScheduleTakePictureRoute, CourseRoute } from '../../constants/screenNames';
import { requestStoragePermission, requestCamera } from '../../services/android_permissions';
import { schoolScheduleStyles as styles, dark_blue, statusBlueColor } from '../../styles';
import updateNavigation from '../NavigationHelper';

/**
 * Permits the user to import their school schedule by selecting or taking a picture or by manual import.
 */
class SchoolSchedule extends React.PureComponent {

	static navigationOptions =  {
		title: 'Add School Schedule',
		headerTransparent: true,
		headerStyle: {
			backgroundColor: 'rgba(0, 0, 0, 0.2)',
		},
	};

	constructor(props) {
		super(props);
		this.state = { 
			containerHeight: null,
		};

		// Updates the navigation location in redux
		updateNavigation('SchoolSchedule', props.navigation.state.routeName);
	}

	/**
	 * In order to open the user's camera roll with permission
	 */
	selectAPicture() {
		if (Platform.OS !== 'ios') {
			requestStoragePermission().then((accepted) => {
				if (accepted) {
					this.props.navigation.navigate(SchoolScheduleSelectPictureRoute);
				}
			});
		} else {
			this.props.navigation.navigate('SchoolScheduleSelectPicture');
		}
	}

	/**
	 * In order to open the user's camera with permission
	 */
	cameraCapture() {
		if (Platform.OS !== 'ios') {
			requestCamera().then((accepted) => {				
				if (accepted) {
					this.props.navigation.navigate(SchoolScheduleTakePictureRoute);
				}
			});
		} else {
			this.props.navigation.navigate(SchoolScheduleTakePictureRoute);
		}
	}

	/**
	 * To go to the appropriate Add Course screen according to the current route
	 */
	manualImport() {
		this.props.navigation.navigate(CourseRoute);
	}

	render() {
		return (
			<View style={styles.container}>
				<StatusBar translucent={true}
					barStyle={Platform.OS === 'ios' ? 'dark-content' : 'default'}
					backgroundColor={statusBlueColor} />

				<View style={styles.content}>
					<View style={styles.instruction}>
						<FontAwesome5 name="university"
							size={130}
							color={dark_blue} />
						<Text style={styles.text}>Import your school schedule by importing or taking a picture</Text>
					</View>
					
					<View style={styles.button}>
						<TouchableOpacity style={styles.buttonSelect}
							onPress={() => this.selectAPicture()}>
							<Text style={styles.buttonSelectText}>SELECT A PICTURE</Text>
						</TouchableOpacity>

						<TouchableOpacity style={styles.buttonTake}
							onPress={() => this.cameraCapture()}>
							<Text style={styles.buttonTakeText}>TAKE A PICTURE</Text>
						</TouchableOpacity>
							
						<Text style={styles.manual}>
							<Text style={styles.textManual}>or import your school schedule </Text>
								
							<Text style={styles.buttonManual}
								onPress={() => this.manualImport()}>manually</Text>

							<Text style={styles.textManual}>.</Text>

						</Text>
					</View>
				</View>
			</View>
		);
	}
}

export default SchoolSchedule;