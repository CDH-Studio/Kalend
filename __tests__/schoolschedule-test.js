import React from 'react';
import renderer from 'react-test-renderer';
import SchoolSchedule from '../src/components/SchoolSchedule';
import {ImageBackground, Image, StatusBar, TouchableOpacity, StyleSheet, Text} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';


test("SchoolSchedule renders correctly", () => {
	const tree = renderer.create(<SchoolSchedule />).toJSON();
	expect(tree).toMatchSnapshot();
});

test("Linear Gradient renders correctly", () => {
	const linearGradient = renderer.create(<LinearGradient style={{flex: 1, width: '100%', height: '110%'}} colors={['#1473E6', '#0E55AA']}></LinearGradient>).toJSON();
	expect(linearGradient).toMatchSnapshot();
});

test("Image Background renders correctly", () => {
	const imageBackground = renderer.create(<ImageBackground  style={{flex: 1, width: '100%', height: '110%'}} source={require('../src/assets/img/loginScreen/backPattern.png')} resizeMode="repeat" ></ImageBackground>).toJSON();
	expect(imageBackground).toMatchSnapshot();
});

test("StatusBar renders correctly", () => {
	const statusBar = renderer.create(<StatusBar translucent={true} backgroundColor={'#00000050'}/>).toJSON();
	expect(statusBar).toMatchSnapshot();
});

test('School icon renders correctly', done => {
	Image.getSize('../assets/img/schoolSchedule/school.png', (width, height) => {
	  const school= renderer.create(<Image style={{height, width}} />).toJSON();
	  expect(school).toMatchSnapshot();
	  done();
	});
});

test("Buttons render correctly", () => {
	const button = renderer.create(<TouchableOpacity style={styles.buttonSelect} onPress={() => this.props.navigation.navigate('SchoolScheduleSelectPicture')}>
	<Text style={styles.buttonSelectText}>SELECT A PICTURE</Text>
</TouchableOpacity>).toJSON();
	expect(button).toMatchSnapshot();
});

test('Active section icon renders correctly', done => {
	Image.getSize('../assets/img/schoolSchedule/sectionActive.png', (width, height) => {
	  const aSection = renderer.create(<Image style={{height, width}} />).toJSON();
	  expect(aSection).toMatchSnapshot();
	  done();
	});
});

test('Inactive section icon renders correctly', done => {
	Image.getSize('../assets/img/schoolSchedule/sectionInactive.png', (width, height) => {
	  const iSection = renderer.create(<Image style={{height, width}} />).toJSON();
	  expect(iSection).toMatchSnapshot();
	  done();
	});
});

const styles = StyleSheet.create({
	container: {
		flex: 1,
		width: '100%',
		height: '130%' //Fixes pattern bug
	},

	content: {
		flex: 1,
		flexDirection: 'column',
		justifyContent: 'space-between',
		marginTop: 160
	},

	schoolIcon: {
		height: 130,
		width: 130
	},

	instruction: {
		flexDirection: 'row',
		justifyContent: 'center'
	},

	text: {
		fontFamily: 'Raleway-Regular',
		color: '#FFFFFF',
		fontSize: 20,
		paddingTop: 30,
		paddingLeft: 15,
		textShadowColor: 'rgba(0, 0, 0, 0.40)',
		textShadowOffset: { width: -1, height: 1 },
		textShadowRadius: 20,
		width: 220
	},

	button: {
		alignItems: 'center'
	},

	buttonSelect: {
		borderRadius: 12,
		backgroundColor: '#FFFFFF',
		padding: 17,
		alignItems: 'center',
		width: 300,
		borderWidth: 3,
		borderColor: '#FFFFFF',
		elevation: 4
	},

	buttonSelectText: {
		fontFamily: 'Raleway-SemiBold',
		fontSize: 15,
		color: '#1473E6',
		
	},

	buttonTake: {
		borderRadius: 12,
		backgroundColor: 'transparent',
		borderWidth: 3,
		borderColor: '#FFFFFF',
		padding: 17,
		alignItems: 'center',
		marginTop: 20,
		width: 300,
		elevation: 4
	},

	buttonTakeText: {
		fontFamily: 'Raleway-SemiBold',
		fontSize: 15,
		color: '#FFFFFF',
		fontWeight:'500',
		textShadowColor: 'rgba(0, 0, 0, 0.40)',
		textShadowOffset: { width: -1, height: 1 },
		textShadowRadius: 20
	},

	section: {
		flexDirection: 'row',
		justifyContent: 'space-between',
		alignItems: 'center',
		marginTop: 50,
		marginBottom: 10
	},

	emptySection: {
		marginLeft: 20,
		opacity: 0
	},

	sectionIconRow: {
		flexDirection: 'row'
	},

	sectionIcon: {
		width: 20,
		height: 20,
		margin: 8
	},

	skipButton: {
		marginRight: 20,
		marginBottom: 2
	},

	skipButtonText: {
		color: 'white',
		fontFamily: 'Raleway-Regular',
		fontSize: 15,
		textShadowColor: 'rgba(0, 0, 0, 0.40)',
		textShadowOffset: {width: -1, height: 1},
		textShadowRadius: 10
	}
});