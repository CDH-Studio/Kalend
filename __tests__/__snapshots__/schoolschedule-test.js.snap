// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Active section icon renders correctly 1`] = `
<Image
  style={
    Object {
      "height": 240,
      "width": 320,
    }
  }
/>
`;

exports[`Buttons render correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderColor": "#FFFFFF",
      "borderRadius": 12,
      "borderWidth": 3,
      "elevation": 4,
      "opacity": 1,
      "padding": 17,
      "width": 300,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#1473E6",
        "fontFamily": "Raleway-SemiBold",
        "fontSize": 15,
      }
    }
  >
    SELECT A PICTURE
  </Text>
</View>
`;

exports[`Image Background renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
      "height": "110%",
      "width": "100%",
    }
  }
>
  <Image
    resizeMode="repeat"
    source={
      Object {
        "testUri": "../../../src/assets/img/loginScreen/backPattern.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "110%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Inactive section icon renders correctly 1`] = `
<Image
  style={
    Object {
      "height": 240,
      "width": 320,
    }
  }
/>
`;

exports[`Linear Gradient renders correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      4279530470,
      4279129514,
    ]
  }
  endPoint={
    Object {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    Object {
      "flex": 1,
      "height": "110%",
      "width": "100%",
    }
  }
/>
`;

exports[`School icon renders correctly 1`] = `
<Image
  style={
    Object {
      "height": 240,
      "width": 320,
    }
  }
/>
`;

exports[`SchoolSchedule renders correctly 1`] = `
<BVLinearGradient
  colors={
    Array [
      4279530470,
      4279129514,
    ]
  }
  endPoint={
    Object {
      "x": 0.5,
      "y": 1,
    }
  }
  locations={null}
  startPoint={
    Object {
      "x": 0.5,
      "y": 0,
    }
  }
  style={
    Object {
      "flex": 1,
      "height": "130%",
      "width": "100%",
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "flex": 1,
        "height": "130%",
        "width": "100%",
      }
    }
  >
    <Image
      resizeMode="repeat"
      source={
        Object {
          "testUri": "../../../src/assets/img/loginScreen/backPattern.png",
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": "130%",
            "width": "100%",
          },
          undefined,
        ]
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-between",
          "marginTop": 160,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "testUri": "../../../src/assets/img/schoolSchedule/school.png",
            }
          }
          style={
            Object {
              "height": 130,
              "width": 130,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Raleway-Regular",
              "fontSize": 20,
              "paddingLeft": 15,
              "width": 220,
            }
          }
        >
          Import your school schedule by importing or taking a picture
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": "#FFFFFF",
              "borderRadius": 12,
              "borderWidth": 3,
              "elevation": 4,
              "opacity": 1,
              "padding": 17,
              "width": 300,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#1473E6",
                "fontFamily": "Raleway-SemiBold",
                "fontSize": 15,
              }
            }
          >
            SELECT A PICTURE
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderColor": "#FFFFFF",
              "borderRadius": 12,
              "borderWidth": 3,
              "elevation": 4,
              "marginTop": 20,
              "opacity": 1,
              "padding": 17,
              "width": 300,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "Raleway-SemiBold",
                "fontSize": 15,
                "fontWeight": "500",
                "textShadowColor": "rgba(0, 0, 0, 0.40)",
                "textShadowOffset": Object {
                  "height": 1,
                  "width": -1,
                },
                "textShadowRadius": 20,
              }
            }
          >
            TAKE A PICTURE
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 10,
            "marginTop": 50,
          }
        }
      >
        <View
          style={
            Object {
              "marginLeft": 20,
              "opacity": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Raleway-Regular",
                "fontSize": 15,
                "textShadowColor": "rgba(0, 0, 0, 0.40)",
                "textShadowOffset": Object {
                  "height": 1,
                  "width": -1,
                },
                "textShadowRadius": 10,
              }
            }
          >
            Skip
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
              "marginRight": -20,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 35,
                },
                Object {
                  "width": 40,
                },
                Object {
                  "fontFamily": "Octicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "rgba(255, 255, 255, 0.50)",
                  "fontSize": 35,
                },
                Object {
                  "width": 40,
                },
                Object {
                  "fontFamily": "Octicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "rgba(255, 255, 255, 0.50)",
                  "fontSize": 35,
                },
                Object {
                  "width": 40,
                },
                Object {
                  "fontFamily": "Octicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "rgba(255, 255, 255, 0.50)",
                  "fontSize": 35,
                },
                Object {
                  "width": 40,
                },
                Object {
                  "fontFamily": "Octicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "rgba(255, 255, 255, 0.50)",
                  "fontSize": 35,
                },
                Object {
                  "width": 40,
                },
                Object {
                  "fontFamily": "Octicons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <View
          style={
            Object {
              "marginBottom": 2,
              "marginRight": 20,
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontFamily": "Raleway-Regular",
                  "fontSize": 15,
                  "textShadowColor": "rgba(0, 0, 0, 0.40)",
                  "textShadowOffset": Object {
                    "height": 1,
                    "width": -1,
                  },
                  "textShadowRadius": 10,
                }
              }
            >
              Skip
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</BVLinearGradient>
`;

exports[`StatusBar renders correctly 1`] = `null`;
